diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/KdocComments.kt

```diff
@@ -45,7 +62,148 @@ class KdocComments(private val configRules: List<RulesConfig>) : Rule("kdoc-comm
             when (node.elementType) {
                 FILE -> checkTopLevelDoc(node)
                 CLASS -> checkClassElements(node)
+                VALUE_PARAMETER -> checkValueParameter(node)
+            }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun checkValueParameter(node: ASTNode) {
+        if (node.parents().none { it.elementType == PRIMARY_CONSTRUCTOR } ||
+                !(node.hasChildOfType(VAL_KEYWORD) || node.hasChildOfType(VAR_KEYWORD))) return
+        val prevComment = if (node.treePrev.elementType == WHITE_SPACE &&
+                (node.treePrev.treePrev.elementType == EOL_COMMENT ||
+                        node.treePrev.treePrev.elementType == BLOCK_COMMENT)) {
+            node.treePrev.treePrev
+        } else if (node.hasChildOfType(KDOC)) {
+            node.findChildByType(KDOC)!!
+        } else if (node.treePrev.elementType == BLOCK_COMMENT || node.treePrev.elementType == EOL_COMMENT) {
+            node.treePrev
+        } else {
+            null
+        }
+        val kDocBeforeClass = node.parent({ it.elementType == CLASS })!!.findChildByType(KDOC)
+        if (prevComment != null) {
```

Comment:
Can't `checkKDocBeforeClass` and `checkBasicKDocBeforeClass` be unified? The difference (as far as I can tell) is that in first case you have text for `@property` tag and need to remove the comment's node, otherwise two methods do the same.

Reply:
I think no, because in `checkKDocBeforeClass` a lot of cases are considered, while ignored in `checkBasicKDocBeforeClass `.

Reply:
ok, let's leave as is